### PR TITLE
[nfc][ubsan-minimal] Refactor error reporting to use a single function

### PR DESCRIPTION
**Pull request by kstoimenov**
This refactoring will allow to make this function weak later on so that it could be overloaded by a client. See #119242.

### CODE REVIEW

File: compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp [hunk 1]
```diff
@@ -20,20 +21,49 @@ static __sanitizer::atomic_uintptr_t caller_pcs[kMaxCallerPcs];
 // that "too many errors" has already been reported.
 static __sanitizer::atomic_uint32_t caller_pcs_sz;
 
-__attribute__((noinline)) static bool report_this_error(uintptr_t caller) {
+#define MSG_PREFIX "ubsan: "
```
Comment (vitalybuka) [inline compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp:24]: Please just inline these macros

**Review by vitalybuka — APPROVED** [review]

File: compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp [hunk 2]
```diff
@@ -1,5 +1,6 @@
 #include "sanitizer_common/sanitizer_atomic.h"
 
+#include <stdio.h>
```
Comment (vitalybuka) [inline compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp:3]: can we avoid the include?

File: compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp (hunk 1 above)
Comment (kstoimenov) [reply compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp]: Done.

File: compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp (hunk 2 above)
Comment (kstoimenov) [reply compiler-rt/lib/ubsan_minimal/ubsan_minimal_handlers.cpp]: Removed.

**Pull request merged**